internal/pubsub/pubsub.go

```diff
@@ -0,0 +1,138 @@
+package pubsub
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+	"regexp"
+
+	cloudpubsub "cloud.google.com/go/pubsub"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/health"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/util"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/run/v1"
+)
+
+// Client represents a client to Google Cloud Pub/Sub.
+type Client interface {
+	Publish(ctx context.Context, msg Message) error
+}
+
+// PubSub is a Google Cloud Pub/Sub client to publish messages.
+type PubSub struct {
+	client *cloudpubsub.Client
+	topic  *cloudpubsub.Topic
+}
+
+// Message is the format when publishing to Pub/Sub.
+type Message struct {
```

Comment:
hmm the best way probably would be to have pkg/notification, and have this be called `RolloutEvent` and pkg/notification/pubsub would be a sub-type of that. 

but for now it's ok